src/extras/render-passes/camera-frame.js

```diff
@@ -0,0 +1,331 @@
+ */
+
+/**
+ * @typedef {Object} Vignette
+ * @property {number} intensity - The intensity of the vignette effect, 0-1 range. Defaults to 0,
+ * making it disabled.
+ * @property {number} inner - The inner radius of the vignette effect, 0-3 range. Defaults to 0.5.
+ * @property {number} outer - The outer radius of the vignette effect, 0-3 range. Defaults to 1.
+ * @property {number} curvature - The curvature of the vignette effect, 0.01-10 range. Defaults to 0.5.
+ */
+
+/**
+ * @typedef {Object} Fringing
+ * @property {number} intensity - The intensity of the fringing effect, 0-100 range. Defaults to 0,
+ * making it disabled.
+ */
+
+/**
+ * @typedef {Object} Taa
+ * @property {boolean} enabled - Whether Taa is enabled. Defaults to false.
+ * @property {number} jitter - The intensity of the camera jitter, 0-1 range. Defaults to 1.
+ */
+
+/**
+ * Implementation of a simple to use camera rendering pass, which supports SSAO, Bloom and
+ * other rendering effects.
+ *
+ * @category Render Pass
+ */
+class CameraFrame {
+    /**
+     * Rendering settings.
+     *
+     * @type {Rendering}
+     */
+    rendering = {
+        renderFormats: [PIXELFORMAT_111110F, PIXELFORMAT_RGBA16F, PIXELFORMAT_RGBA32F],
+        stencil: false,
+        renderTargetScale: 1.0,
+        samples: 1,
+        sceneColorMap: false,
+        sceneDepthMap: false,
+        toneMapping: 0,
+        sharpness: 0.0
+    };
+
+    /**
+     * SSAO settings.
+     *
+     * @type {Ssao}
+     */
+    ssao = {
+        type: SSAOTYPE_NONE,
+        blurEnabled: true,
+        intensity: 0.5,
+        radius: 30,
+        samples: 12,
+        power: 6,
+        minAngle: 10,
+        scale: 1
+    };
+
+    /**
+     * Bloom settings.
+     *
+     * @type {Bloom}
+     */
+    bloom = {
+        intensity: 0,
+        lastMipLevel: 1
+    };
+
+    /**
+     * Grading settings.
+     *
+     * @type {Grading}
+     */
+    grading = {
+        enabled: false,
+        brightness: 1,
+        contrast: 1,
+        saturation: 1,
+        tint: new Color(1, 1, 1, 1)
+    };
+
+    /**
+     * Vignette settings.
+     *
+     * @type {Vignette}
+     */
+    vignette = {
+        intensity: 0,
+        inner: 0.5,
+        outer: 1,
+        curvature: 0.5
+    };
+
+    /**
+     * Taa settings.
+     *
+     * @type {Taa}
+     */
+    taa = {
+        enabled: false,
+        jitter: 1
+    };
+
+    /**
+     * Fringing settings.
+     *
+     * @type {Fringing}
+     */
+    fringing = {
+        intensity: 0
+    };
+
+    options = new CameraFrameOptions();
+
+    /**
+     * @type {RenderPassCameraFrame|null}
+     * @private
+     */
+    renderPassCamera = null;
+
+    /**
+     * Creates a new CameraFrame instance.
+     *
+     * @param {AppBase} app - The application.
+     * @param {CameraComponent} cameraComponent - The camera component.
+     */
+    constructor(app, cameraComponent) {
```

Comment:
How come the `CameraFrame` takes a `cameraComponent` instead of doing:

```
entity.camera.frame = new CameraFrame();
```

Reply:
CameraFrame is a module used to simplify set up for the user, and manages and assigns render passes to the camera component. Camera component works directly with render passes, not with CameraFrame, as there can be other implementations of a helper class like this, using different frame setup.